### PR TITLE
ci: cache deps

### PR DESCRIPTION
**Pull request by nic-chen**
### What this PR does / why we need it:

Cache the deps directory in ci.

The key of the cache is the hash value of the file content of rockspec/apisix-master-0.rockspec. 

If the hash value is changed, the cache will not be used. 

The cache will become invalid if it is not used within 7 days.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: .github/workflows/centos7-ci.yml [hunk 1]
```diff
@@ -26,100 +26,108 @@ jobs:
...
-      run: |
-        docker exec centos7Instance bash -c "cd apisix && ./utils/centos7-ci.sh run_case"
-
-    - name: Publish Artifact
-      if: ${{ startsWith(github.ref, 'refs/heads/release/') }}
-      uses: actions/upload-artifact@v2.2.3
-      with:
-        name: "rpm"
-        path: "./apisix-build-tools/output/apisix-${{ steps.branch_env.outputs.version }}-0.x86_64.rpm"
+      - name: Check out code
```
Comment (spacewander) [inline .github/workflows/centos7-ci.yml:29]: Why change the indentation of the existing code?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -56,6 +56,14 @@ jobs:
         with:
           submodules: recursive
 
+      - name: Cache deps
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-deps
+        with:
+          path: ~/deps
```
Comment (spacewander) [inline .github/workflows/build.yml:64]: The `~/deps` is defined but not used?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (nic-chen) [reply .github/workflows/build.yml]: it's a  typo, fixed. thanks.

File: .github/workflows/centos7-ci.yml (hunk 1 above)
Comment (nic-chen) [reply .github/workflows/centos7-ci.yml:29]: The original format is not standard, I think it should be fixed. Do I need to revert it?

Comment (spacewander) [reply .github/workflows/centos7-ci.yml:29]: Better to avoid irrelevant change.

**Comment by spacewander** [comment]
The cache code can be removed now:
https://github.com/apache/apisix/blob/42ea885fb76400401f28f49cffcb2e54ba87ec49/ci/common.sh#L36-L41

**Comment by nic-chen** [comment]
@spacewander  updated. please have a look again when you have time. thanks.

**Comment by spacewander** [comment]
The CI failed:
```
++ mv grpc_server_example build-cache/
mv: cannot move 'grpc_server_example' to 'build-cache/': Not a directory
```

**Comment by spacewander** [comment]
Look like something wrong is cached?
https://github.com/apache/apisix/pull/4026/checks?check_run_id=2329191183

**Review by spacewander — APPROVED** [review]

**Review by totemofwolf — APPROVED** [review]
LTME

**Pull request merged**